### DISCUSSION TITLE
Migrating Next app from Full SSR to hybrid SSR/SSG - When should we use "getInitialProps"?

### DISCUSSION THREAD

**Discussion by Vadorequest** [vercel/next.js#12215] [Help] [answered]
Since 9.3, 3 new utility functions are available.

https://nextjs.org/docs/basic-features/data-fetching

But `getInitialProps` is mentioned nowhere. Is it meant not to be used anymore (in newer projects)? Or is it still useful for some use cases?

I'm in the process of migrating a Next.js app that uses `getInitialProps` in `_app` (SSR enabled for every page) to a hybrid app with per-page SSG/SSR. I'm not exactly sure what to do.

Also, does `_app` supports `getStaticProps`, or must it be added for each page? It's not stated in the doc.

**Comment by brianlovin** [comment, marked as answer]
`getInitialProps` can be used still (legacy), but I believe it's preferred to use `getServerSideProps` (where the behaviors are loosely the same). As you find places where you would be able to use `getStaticProps`, that is even better.

Re: using `getStaticProps` in `_app`, I did a quick search and found this: https://github.com/zeit/next.js/discussions/10949
